src/plugins/saved_objects/public/saved_object/helpers/create_source.ts

```diff
@@ -80,3 +80,44 @@ export async function createSource(
     return await Promise.reject(err);
   }
 }
+
+export async function createSourceUtil(
```

Comment:
This helper makes a lot of sense to me, but the name is a little confusing to me - isn't this just `saveWithConfirmation` (or something like that)?

Also we shouldn't make this helper dependent on the legacy type `SavedObject` - just pass in the stuff you need directly.

Reply:
Of course, it was just a draft function name.
Updated.